### ISSUE TITLE
How to load keras model which has Universal Sentence Encoder embedding layer once then use it for prediction?

### ISSUE THREAD

**Issue by keduadoi** [tensorflow/hub#563]
I'm a newbie in tensorflow. Currently I'm creating a model for classification. My code saved keras model as h5 file successfully, now I'm trying to use this model for predict any input text's label from REST API.

This is my current code and it works

def universal_embedding(x):
  embed = hub.Module(url)
  return embed(tf.squeeze(tf.cast(x, tf.string), axis=[1]), signature="default", as_dict=True)["default"]

def create_model():
  input_text = layers.Input(shape=(1,), dtype=tf.string)
  embedding = layers.Lambda(universal_embedding, output_shape=(512,))(input_text)
  dense = layers.Dense(256, activation='relu')(embedding)
  pred = layers.Dense(4, activation='softmax')(dense)

  model = Model(inputs=[input_text], outputs=pred)
  model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
  model.summary()

  return model

@app.route("/predictLabel", methods=['GET', 'POST'])
@cross_origin()
def predict_label():
  model = create_model()

  request_data = json.loads(request.data)
  text = request_data['text']

  request_list = [text]

  request_list = np.array(request_list, dtype=object)[:, np.newaxis]

  with tf.compat.v1.Session() as session:
     K.set_session(session)
     session.run([tf.compat.v1.global_variables_initializer(), tf.compat.v1.tables_initializer()])
     model.load_weights('./model.h5')
     predicts = model.predict(request_list, batch_size=32)

As you can see, every a new request comes, the model needs to load again, so how can I move the loading model part outside the function so the application loads it when starting once? I know we need to deal with tensorflow session and graph to do this but not sure how to implement it. Thank you very much.

**Comment by akhorlin** [comment]
One options is to use [TF Serving API](https://www.tensorflow.org/tfx/guide/serving) for serving models.

**Comment by gowthamkpr** [comment]
@keduadoi Use the Tensorflow Serving API as mentioned above and it should solve your problem.

**Comment by swati210994** [comment]
What is the Model in your case?
this line
model = Model(inputs=[input_text], outputs=pred)
